### ISSUE TITLE
Use const enums for BindingScope, BindingType, TargetType

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [inversify/InversifyJS#706]
Now that TypeScript supports string values for enums, it'd be very convenient to switch from this:

```typescript
type BindingScope = "Singleton" | "Transient" | "Request";
type BindingType = "ConstantValue" | "Constructor" | "DynamicValue" | "Factory" | "Function" | "Instance" | "Invalid" | "Provider";
type TargetType = "ConstructorArgument" | "ClassProperty" | "Variable";    
```

...to this:

```typescript
const enum BindingScope {
    Singleton = "Singleton",
    Transient = "Transient",
    Request = "Request",
}

const enum BindingType {
    ConstantValue = "ConstantValue",
    Constructor = "Constructor",
    DynamicValue = "DynamicValue",
    Factory = "Factory",
    Function = "Function",
    Instance = "Instance",
    Invalid = "Invalid",
    Provider = "Provider",
}

const enum TargetType {
    ConstructorArgument = "ConstructorArgument",
    ClassProperty = "ClassProperty",
    Variable = "Variable",
}
```

That way, instead of hardcoding string literals:

```typescript
const container = new Container({
    defaultScope: "Singleton",
});
```

...we can use slightly-easier-to-get-intellisense-for enum members:

```typescript
const container = new Container({
    defaultScope: BindingScope.Singleton,
});
```

**Comment by remojansen** [comment]
This change looks good to me. Are u planning to send a PR?

**Comment by JoshuaKGoldberg** [comment]
Yes :)

**Comment by JoshuaKGoldberg** [comment]
You know, this actually is a breaking change. You would no longer be able to use sring literals as many likely do. I'll hold off the change.

https://github.com/Microsoft/TypeScript/issues/17690
https://github.com/Microsoft/TypeScript/issues/3192#issuecomment-261720275

Steps to make the change locally are:
* Delete `literal_types.ts`
* Perform the OP's suggested replacement in `interfaces.ts`
* Replace usages of the enum types exported from that file with the newly exported enums in `interfaces.ts`

**Comment by JoshuaKGoldberg** [comment]
https://github.com/Microsoft/TypeScript/issues/17690#issuecomment-350889463

Will just use a `const` object.

**Comment by remojansen** [comment]
@JoshuaKGoldberg I noticed that we already have them https://github.com/inversify/InversifyJS/blob/e3b2871729aeb328ef5d2366282bffa4dc2c1a62/src/constants/literal_types.ts but they are [not exposed](https://github.com/inversify/InversifyJS/blob/e3b2871729aeb328ef5d2366282bffa4dc2c1a62/src/inversify.ts)

**Comment by remojansen** [comment]
Closing this issue by https://github.com/inversify/InversifyJS/pull/718

You can now do:

```ts
import { BindingScopeEnum, Container  } from "inversify";

const container = new Container({
    defaultScope: BindingScopeEnum.Singleton,
});
```

This feature is available in 4.7.0 :tada:
